qa/workunits/ceph-helpers.sh

```diff
@@ -1267,6 +1267,48 @@ function test_wait_for_clean() {
 #######################################################################
 
 ##
+# Wait until the cluster becomes HEALTH_OK again or if it does not make progress
+# for $TIMEOUT seconds.
+#
+# @return 0 if the cluster is HEALTHY, 1 otherwise
+#
+function wait_for_health() {
+    local grepstr=$1
+    local -a delays=($(get_timeout_delays $TIMEOUT .1))
+    local -i loop=0
+
+    while ! ceph health detail | grep "$grepstr" ; do
+        ceph health detail
+	if (( $loop >= ${#delays[*]} )) ; then
+            return 1
+        fi
+        sleep ${delays[$loop]}
+        loop+=1
+    done    
+    return 0
+}
+
+function wait_for_health_ok() {
+     wait_for_health "HEALTH_OK" || return 1
+     return 0
+} 
+
+function test_wait_for_health_ok() {
+    local dir=$1
+
+    setup $dir || return 1
+    run_mon $dir a --osd_pool_default_size=1 || return 1
+    run_mgr $dir x || return 1
+    ! TIMEOUT=1 wait_for_health_ok || return 1
+    run_osd $dir 0 || return 1
+    wait_for_health_ok || return 1
```

Comment:
There will never be a HEALTH_OK at this point with only 1 osd and no pools unless you tweak configuration.   This is causing the failed Jenkins run.

HEALTH_ERR too few PGs per OSD (4 < min 30); Full ratio(s) out of order
too few PGs per OSD (4 < min 30)
osd_failsafe_full_ratio (0.97) < full_ratio (0.99), increased

You need to set "mon_pg_warn_min_per_osd" config to false and "osd_failsafe_full_ratio" to .99 at least in the mon.